### PR TITLE
BIA-676 Add StudentProgramDim view to collection

### PR DESCRIPTION
**Pull request by bptillman**
This adds a new StudentProgramDim view to the analytics base collection so that going forward, we can query it instead of querying the ODS directly.
Tests have been added for when a record does and does not exist in the view along with testing when a student is in multiple programs ensuring that multiple rows come back from the view. If any additional tests should be added please let me know.

### CODE REVIEW

**Review by jleiva-gap — CHANGES_REQUESTED** [review]
DS 2. If I comment out the following lines, tests should failed, but all test are green.

1. 25
2. 26
3. 27

    DS 3. If I comment out the following lines, tests should failed, but all test are green.

1. 25
2. 26
3. 27

You need to add more records so that the join conditions fail if any are removed in the view script.

**Comment by bptillman** [comment]
> 
> 
> ```
> DS 2. If I comment out the following lines, tests should failed, but all test are green.
> ```
> 
>     1. 25
> 
>     2. 26
> 
>     3. 27
>        DS 3. If I comment out the following lines, tests should failed, but all test are green.
> 
>     4. 25
> 
>     5. 26
> 
>     6. 27
> 
> 
> You need to add more records so that the join conditions fail if any are removed in the view script.

I have updated each of the data inserts to insert an additional program/org so that when the and conditions are commented out in the join, the view will produce extra results.

File: src/EdFi.AnalyticsMiddleTier.DataStandard2/Base/MSSQL/0025-View-StudentProgramDim-Create.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+GO
+CREATE VIEW [analytics].[StudentProgramDim]
+AS
+    SELECT [BeginDate],program.[EducationOrganizationId],program.[ProgramName],student.[StudentUSI],
+    CONCAT (Student.StudentUniqueId,'-',StudentSchoolAssociation.SchoolId) AS StudentSchoolKey
+    FROM [edfi].[Program]
+    INNER JOIN
+        [edfi].[StudentProgramAssociation] StudentProgram
+        ON StudentProgram.ProgramName = Program.ProgramName
+        AND StudentProgram.ProgramTypeId = Program.ProgramTypeId
```
Comment (jleiva-gap) [inline src/EdFi.AnalyticsMiddleTier.DataStandard2/Base/MSSQL/0025-View-StudentProgramDim-Create.sql:26]: If I comment out this line, test are not failing. More rows are needed to validate this JOIN (e.g. Add a new program with different ProgramTypeId, but with same ProgramName and EducationOrganizationId of other row.

File: src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0023-View-StudentProgramDim-Create.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+GO
+CREATE VIEW [analytics].[StudentProgramDim]
+AS
+    SELECT [BeginDate],program.[EducationOrganizationId],program.[ProgramName],student.[StudentUSI],
+    CONCAT (Student.StudentUniqueId,'-',StudentSchoolAssociation.SchoolId) AS StudentSchoolKey
+    FROM [edfi].[Program]
+    INNER JOIN
+        [edfi].[StudentProgramAssociation] StudentProgram
+        ON StudentProgram.ProgramName = Program.ProgramName
+        AND StudentProgram.ProgramTypeDescriptorId = Program.ProgramTypeDescriptorId
```
Comment (jleiva-gap) [inline src/EdFi.AnalyticsMiddleTier.DataStandard31/Base/MSSQL/0023-View-StudentProgramDim-Create.sql:26]: If I comment out this line, test are not failing. More rows are needed to validate this JOIN (e.g. Add a new program with different ProgramTypeDescriptorId, but with same ProgramName and EducationOrganizationId of other row.

File: src/EdFi.AnalyticsMiddleTier.DataStandard32/Base/MSSQL/0023-View-StudentProgramDim-Create.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+GO
+CREATE VIEW [analytics].[StudentProgramDim]
+AS
+    SELECT [BeginDate],program.[EducationOrganizationId],program.[ProgramName],student.[StudentUSI],
+    CONCAT (Student.StudentUniqueId,'-',StudentSchoolAssociation.SchoolId) AS StudentSchoolKey
+    FROM [edfi].[Program]
+    INNER JOIN
+        [edfi].[StudentProgramAssociation] StudentProgram
+        ON StudentProgram.ProgramName = Program.ProgramName
+        AND StudentProgram.ProgramTypeDescriptorId = Program.ProgramTypeDescriptorId
```
Comment (jleiva-gap) [inline src/EdFi.AnalyticsMiddleTier.DataStandard32/Base/MSSQL/0023-View-StudentProgramDim-Create.sql:26]: If I comment out this line, test are not failing. More rows are needed to validate this JOIN (e.g. Add a new program with different ProgramTypeDescriptorId, but with same ProgramName and EducationOrganizationId of other row.

**Review by jleiva-gap — CHANGES_REQUESTED** [review]

**Review by jleiva-gap — APPROVED** [review]

**Pull request merged**